suse2021-ns/xhtml/xref.xsl

```diff
@@ -125,81 +125,97 @@
 
 
 <xsl:template match="d:xref" name="xref">
-  <xsl:variable name="targets" select="key('id',@linkend)"/>
-  <xsl:variable name="target" select="$targets[1]"/>
-  <xsl:variable name="refelem" select="local-name($target)"/>
-  <xsl:variable name="target.book" select="($target/ancestor-or-self::d:article|$target/ancestor-or-self::d:book)[1]"/>
-  <xsl:variable name="this.book" select="(ancestor-or-self::d:article|ancestor-or-self::d:book)[1]"/>
-  <xsl:variable name="lang" select="(ancestor-or-self::*/@lang|ancestor-or-self::*/@xml:lang)[1]"/>
+  <xsl:variable name="context" select="."/>
+  <xsl:variable name="target" select="key('id', @linkend)[1]"/>
   <xsl:variable name="xref.in.samebook">
     <xsl:call-template name="is.xref.in.samebook">
       <xsl:with-param name="target" select="$target"/>
     </xsl:call-template>
   </xsl:variable>
 
-  <xsl:call-template name="check.id.unique">
-    <xsl:with-param name="linkend" select="@linkend"/>
-  </xsl:call-template>
-
- <!--
+   <!--
     The xref resolution is a bit tricky. We need to distinguish these cases:
 
     1. if $rootid = '' -> use original code
     2. if $rootid != '' and rootid points to the root node -> use original code
-    3. if (1) or (2) does not apply -> reference into another book
-    4. if we point to another book AND we have an @xrefstyle -> output the
+    3. if we point to another book AND we have an @xrefstyle -> output the
        same text content as the original xref template but remove the link
+    4. otherwise -> reference into another book
  -->
- <xsl:choose>
-  <xsl:when test="$rootid = ''">
-   <xsl:apply-imports/>
-  </xsl:when>
-  <xsl:when test="$rootid != '' and $xref.in.samebook != 0">
-   <xsl:apply-imports/>
-  </xsl:when>
-  <!-- If we point to another book AND we have an xrefstyle present,
-       skip creating intra-xrefs and use the default, but without
-       a clickable text.
-    -->
-  <xsl:when test="$xref.in.samebook = 0 and @xrefstyle">
-   <xsl:variable name="rtf">
-    <xsl:apply-imports/>
-   </xsl:variable>
-   <xsl:variable name="node" select="exsl:node-set($rtf)/*"/>
-
-   <span class="intraxref">
-    <xsl:copy-of select="$node/node()"/>
-   </span>
-  </xsl:when>
-  <xsl:when test="$xref.in.samebook != 0 or
-                  /d:set/@id=$rootid or
-                  /d:article/@id=$rootid">
-   <!-- An xref that stays inside the current book or when $rootid
+
+  <xsl:call-template name="check.id.unique">
+    <xsl:with-param name="linkend" select="@linkend"/>
+  </xsl:call-template>
+
+  <xsl:message>xref:
+               rootid=<xsl:value-of select="$rootid"/>
+              linkend=<xsl:value-of select="@linkend"/>
+  is.xref.in.samebook=<xsl:value-of select="$xref.in.samebook"/>
+  </xsl:message>
+
+    <xsl:choose>
```

Comment:
Something's definitely wrong with whatever is indenting your code. :)

Reply:
Yes, that was unfortunate. I seems I was a bit hasty by adding you and pushing the (unfinished) code at the same time. I did it on Friday to have at least a PR.